### PR TITLE
3.6 native plugin loader

### PR DESCRIPTION
**Pull request by PatriceJiang**
Re: https://github.com/cocos/3d-tasks/issues/11441

### Changelog

* 

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.

  > Manual trigger with `@cocos-robot run test cases` afterward.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by PatriceJiang** [comment]
### Process of importing a plug-in into a project 

1. unzip and copy the plug-in to any subdirectory of the `extensions/` or `native/` directory. The plug-in directory structure is as follows:



image

image


The plug-in configuration content is as follows

image


2. Run `Build`

The user's project will generate the compilation target `plugin_register`

image

And plug-in cpp source file

image

Generate cmake configuration file at the same time

image


> If no plug-in found, these steps will be skipped.

3. Disable plug-in

Modify the corresponding configuration and disable the plug-in and it can be operated manually or modified by the editor plug-in

image


Take effect after re-build.

File: native/cmake/scripts/plugin_support/plugin_cfg.js [hunk 1]
```diff
@@ -0,0 +1,846 @@
+/*
+ * Generated by PEG.js 0.10.0.
```
Comment (minggo) [inline native/cmake/scripts/plugin_support/plugin_cfg.js:2]: Please use `-` to connect words for folder names.

File: native/cocos/plugins/bus/EventBus.h [hunk 2]
```diff
@@ -0,0 +1,285 @@
+#pragma once
```
Comment (minggo) [inline native/cocos/plugins/bus/EventBus.h:25]: Please add license.

File: native/cocos/plugins/Plugins.h [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+****************************************************************************/
+#pragma once
+
+#include "plugins/bus/EventBus.h"
+
+#if _WIN32
+    #define CC_PLUGIN_DLL_EXPORT __declspec(dllexport)
```
Comment (dumganhar) [inline native/cocos/plugins/Plugins.h:30]: Non-windows platforms should also define this macro, for example, gcc & clang compiler need `__attribute__((visibility("default")))`

File: native/cocos/plugins/bus/EventBus.cpp [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+Listener::Listener(EventBus *bus) : _bus(bus) {
+    bus->addListener(this);
+}
+
+Listener::~Listener() {
+    _bus->removeListener(this);
+}
+
+EventBus *EventBus::accquire(BusType type) {
```
Comment (dumganhar) [inline native/cocos/plugins/bus/EventBus.cpp]: Typo ? accquire -> acquire ?

**Review by dumganhar — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Pull request merged**